src/tools/interop/interop.cpp

```diff
@@ -960,14 +971,25 @@ QUIC_THREAD_CALLBACK(InteropTestCallback, Context)
         }
         if (TestResults[TestContext->EndpointIndex].Alpn == nullptr) {
             TestResults[TestContext->EndpointIndex].Alpn = Alpn;
-            Alpn = nullptr;
         }
         QuicLockRelease(&TestResultsLock);
     } else {
         TestFailed = true;
+        ThisTestFailed = true;
     }
 
-    free((void*)Alpn);
+    QuicTraceLogInfo(
+        InteropTestStop,
+        "[ntrp] Test Stop, Server: %s, Port: %hu, Tests: 0x%x, Negotiated Alpn: %s, Passed: %s.",
+        PublicEndpoints[EndpointIndex].ServerName,
+        TestContext->Port,
+        (uint32_t)TestContext->Feature,
+        Alpn,
+        ThisTestFailed ? "false" : "true");
+
+    if (ThisTestFailed) {
+        free((void*)Alpn);
+    }
```

Comment:
Unnecessary if check. `free` allows `NULL` pointers to be passed in.

Reply:
I'm not setting it to NULL anymore, since the logging function needs to read it.

Reply:
Ok.